plugins/cybereason/icon_cybereason/util/api.py

```diff
@@ -202,3 +230,23 @@ def generate_sensor_filter(identifier: str) -> dict:
             sensor_filter["filters"][0]["fieldName"] = "machineName"
 
         return sensor_filter
+
+    @staticmethod
+    def get_machine_targets(results: str, machine_guid: str) -> list:
+        target_ids = []
+
+        for key, value in results.items():
+            try:
+                for i in value["elementValues"]["ownerMachine"]["elementValues"]:
+                    if machine_guid in i["guid"]:
+                        target_ids.append(key)
+            except KeyError:
```

Comment:
Small possible change- have some way of logging a warning? In what cases should there be a result but not have the corresponding keys? If there is a simple answer to that, just leave a comment in the code please :)

Reply:
Hey @pmara-r7,
There was some cases where the response from Cybereason had `elementValues` list but with an empty object in it. In this case the `i["guid"]` would raise a `KeyError` as that key would not be available. I didn't feel like logging a warning is necessary as we raise an exception just couple of lines below. I'll add a comment with that info to the code.

Reply:
Just in terms of future maintainability, it would be great to put the KeyError where it actually is expected to throw the error- for example, if there is no "ownerMachine" key in a future version of the API, this code would just keep returning the same PluginException, which would be difficult for us to diagnose as an API change when it came up. One way around this it to put the existing try catch inside the for loop. Then you could further wrap the entire for loop in a try catch where the catch lets users/developers know that the API returned unexpected key-value pairs (aka if "elementValues", "ownerMachine", or "elementValues" keys change in future versions) .